GameProject/Assets/Turn based assets/Scripts/ISelection.cs

```diff
@@ -0,0 +1,11 @@
+﻿using UnityEngine;
+
+namespace Turn_based_assets.Scripts
```

Comment:
You have a namespace problem. Should be `TurnBasedAssets`, which means you will need to also change the folder name to `TurnBasedAssets` also to keep it in-convention and ensure to update this everywhere it's used.